### ISSUE TITLE
Incorrect parsing error for TSQL

### ISSUE THREAD

**Issue by james-camacho-tel360** [sqlfluff/sqlfluff#6152]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I'm trying to run `sqlfluff fix` on a file that creates a SQL user but I get a "found unparsable section" on the `for login [login]` portion of the code. The code works in my Azure SQL DB and is in the documentation for how to create a user via TSQL

The username is an email associated with a Microsoft Entra ID, so because of the special character it needs to be enclosed in `[]`

### Expected Behaviour

No parsing error flagged

### Observed Behaviour

```
L:   9 | P:  33 |  PRS | Line 9, Position 33: Found unparsable section: 'for
                       | login [rponce@tel360.com];\n\n-- allow...'
```

### How to reproduce

Test SQL 
```
if
    exists (
        select 1 from sys.database_principals where name = 'myuser'
    )
    begin
        exec ('drop user [myuser])
    end;

create user [myuser] for login [myuser];
```

### Dialect

TSQL

### Version

2.3.5

### Configuration

```
[sqlfluff]
dialect = tsql
exclude_rules = ST06, RF05, CP02
warnings = LT05
```

The line also had a `--noqa: PRS` comment but still was flagged when `sqlfluff fix` was run

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Two observation: You have an _unclosed_ string in your example, which is resulting the unparsable error:

```sql
 exec ('drop user [myuser]) 
```

Once you close the string the example can be parsed properly with the latest `sqlfluff`, as #6608 added a proper support for `CREATE USER` in the meantime.

I'd suggest to close the issue.
